### PR TITLE
Replace waitFor(() => {}) with act()

### LINKED ISSUE #6384: [Tech Debt] Replace waitFor(() => {}) with act()
**Is your feature request related to a problem? Please describe.**
Some tests in our code base are using `waitFor(() => {})` with an empty callback which is considered bad practice.

Removing the wait() from these tests reveals warnings/errors where we should be using `act()` to ensure updates to the toggles and MostRead containers are handled correctly, `waitFor()` was masking these warnings.

```
Warning: An update to CanonicalMostRead inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
    
    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://fb.me/react-wrap-tests-with-act
```

**Describe the solution you'd like**
Replace any use of waitFor(() => {}) that has an empty callback with act() https://reactjs.org/docs/test-utils.html#act

And ensure the test continue to pass successfully, this time without any warnings.

**Describe alternatives you've considered**
N/A

**Testing notes**
Unit tests should all pass.

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #6384

**Overall change:**
Replace any use of waitFor(() => {}) that has an empty callback with act() https://reactjs.org/docs/test-utils.html#act

**Code changes:**
- Replace `waitFor(() => {})) ` with `act()` in the FrontPage tests

---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] I have added labels to this PR for the relevant pod(s) affected by these changes
- [X] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by EinsteinNjoroge — COMMENTED** [review]
Shouldn't we replace this instance of `waitFor` as well? https://github.com/bbc/simorgh/blob/afc88bce1c0d4ddc84c178dbf48bbf8454136c71/src/app/pages/FrontPage/index.test.jsx#L139

File: src/app/pages/FrontPage/index.test.jsx [hunk 1]
```diff
@@ -144,9 +144,11 @@ describe('Front Page', () => {
 
   describe('Assertions', () => {
     it('should render visually hidden text as h1', async () => {
-      const { container } = render(
-        <FrontPageWithContext pageData={pageData} />,
-      );
+      let container;
+      await act(async () => {
```
Comment (tochwill) [inline src/app/pages/FrontPage/index.test.jsx:146]: This pattern is much nicer!

**Review by tochwill — DISMISSED** [review]

**Comment by DenisHdz** [comment]
> Shouldn't we replace this instance of `waitFor` as well?
> 
> https://github.com/bbc/simorgh/blob/afc88bce1c0d4ddc84c178dbf48bbf8454136c71/src/app/pages/FrontPage/index.test.jsx#L139

This `waitFor` should be fine. It was added recently in [this](https://github.com/bbc/simorgh/pull/6372) PR, because `wait` was deprecated https://testing-library.com/docs/dom-testing-library/api-async#wait-deprecated-use-waitfor-instead.

However, I'm happy to replace it with `act` to keep the file consistent.

**Review by tochwill — APPROVED** [review]

**Review by thekp — APPROVED** [review]
The changes look good to me. 

But from what I understand from the issue, should we be changing all occurrences of waitFor() in Simorgh to be using the `act()` pattern. 

If so, I think we can merge this PR but keep the original issue open.

**Comment by DenisHdz** [comment]
> The changes look good to me.
> 
> But from what I understand from the issue, should we be changing all occurrences of waitFor() in Simorgh to be using the `act()` pattern.
> 
> If so, I think we can merge this PR but keep the original issue open.

@thekp The issue is just about replacing the `waitFor`s  that have an **empty callback** (`waitFor(() => {})` because it is considered a bad practice.

However, we can create another issue to look into replacing all of them with `act`s.

**Comment by thekp** [comment]
> @thekp The issue is just about replacing the waitFors that have an empty callback (waitFor(() => {}) because it is considered a bad practice.

ahh I see, thanks for clarifying.

**Pull request merged**